### ISSUE TITLE
SA1305 for parameters

### ISSUE THREAD

**Issue by Trass3r** [DotNetAnalyzers/StyleCopAnalyzers#1928]
```cs
public void foo(bool bDelete)
{
	// this one is caught but not the parameter
	bool bDoSomething;
}
```

**Comment by sharwell** [comment]
Does your method override another method in a base class or implement an interface method? I believe this warning is disabled for parameters in those cases.

**Comment by Trass3r** [comment]
No it's an NUnit Test method.

**Comment by Noryoko** [comment]
The analyzer for SA1305 only looks at fields and local variable declarations. Parameters are not covered. We should look at classic to see if this is a deviation in behavior.

**Comment by sharwell** [comment]
@Noryoko we should also make sure it covers all local variables. I had to update SA1312 recently because a bunch of them were missed.

**Comment by Noryoko** [comment]
I'll take a look this evening.

**Comment by sharwell** [comment]
The issue and fix for SA1312 are in #1856 and #1864, respectively. :smile:

**Comment by sharwell** [comment]
@Noryoko I filed issue #1930 to address the local variables issue.

**Comment by Noryoko** [comment]
Parameters are checked by classic. Inherited methods generate warnings on both the child and parent.

I think we should ignore any inherited methods.

**Comment by sharwell** [comment]
> I think we should ignore any inherited methods.

I agree. At the same time as this is implemented, we should document this deviation in KnownChanges.md.

**Comment by bjornhellander** [comment]
I guess the reason for excluding overridden methods is "It's not my fault. I only redefined an existing method!", right? In that case, should this rule only exempt parameters which has the same name in the original definition (or one of them, since there might be more than one)?

For example, I see no reason to let this pass without a diagnostic:
```
class A
{
    public void DoStuff(long value) {}
}

class B : A
{
    public void DoStuff(long lValue) {}
}
```

**Comment by sharwell** [comment]
@bjornhellander I agree.

**Comment by vweijsters** [comment]
I would suggest leaving overriding methods (and interface implementing methods) alone, as these are in the domain of CA1725 (from roslyn-analyzers) and/or SA1315 (if we decide to keep it). 

Flagging the base definition should be the only one for this rule I think. The code fix will be interesting, as it will also have to rename all overriding instances to prevent triggering CA1725 / SA1315.

**Comment by bjornhellander** [comment]
OK, I didn't realize it was bordering other rules. Simple is good too! :smile:

Since this rule doesn't even have a code fix as of today, as far as I can see, would it be ok to handle that in a separate issue?
